docs/content/basics/testing_theory.md

```diff
@@ -1,57 +1,163 @@
+
+3. **Shared tests**: These are tests that are written once, but can run either as JVM or as Instrumentation tests. The principle is simple: *Write once, run everywhere*.
+   These tests contain Android-specific code. Such code would need a device to run on. In order to enable that, when running on the JVM, that code is replaced by mocks under the hood. We'll go deeper into this later on.
+
+## Classifying tests by their target
+Depending on the target of our tests, they are split into the following categories
+
+1. **UI tests**: focus on testing the WHAT (interaction/navigation); *WHAT is displayed* when interacting with the screen elements.
+2. **Screenshot/Snapshot tests**:  focus on testing the HOW (visuals): *HOW a view is displayed* under a given state or configuration.
+3. **Non-UI tests**: focus on testing non-ui related code like *BUSINESS LOGIC* or *DATABASE MIGRATIONS* among others
+
+### Should I write Snapshot tests or Ui Tests?
+It might be confusing to understand when to write Ui test rather than Snapshot tests and vice versa. They do not replace each other. Their focus is different as previously mentioned. So let's imagine the following screen, which is a RecyclerView
+
+![gif300Faster.gif](https://cdn.hashnode.com/res/hashnode/image/upload/v1631992848600/ajg2fTGRG.gif)
+
+A *Ui test would verify*, e.g. that after deleting a row in the recycler View, that row is not displayed anymore. It would test *WHAT is displayed after interacting with the view*.
+
+On the other hand, a *snapshot test would verify HOW that row is displayed* under numerous states and configurations: e.g. dark/light mode, LTR/RTL languages, different font sizes, wide/narrow screens...
+
+You might think you could write a full screen snapshot test after deleting the row to verify it is not displayed anymore. Although it might work at the beginning, this approach comes with its own problems, the most relevant is that your **tests become brittle**.
+
+Let me explain that with an example.
+Imagine you change how the row is displayed. If you verify the snapshot test, it will fail. You need to record a new snapshot including those changes on the row. The issue here is that the focus of your test was to verify that the deleted row is not displayed anymore. What does it have to do with changing the appearance of the row?
+
+Therefore, *every subtle change on the screen will require to record a new snapshot, although that change had nothing to do with the initial scope of the test*.
+On the other hand, a Ui test would have not failed since we would be asserting whether the deleted row was displayed or not. No visuals involved.
+
+To sum it up: **use both Ui test and Snapshot tests**, they complement each other. Think which is the right tool for the task at hand.
+
+### Shared tests
```

Comment:
Looks like a page/section/subsection on its own?

Reply:
from my side I guess we should separate it to another page